src/odemis/acq/align/autofocus.py

```diff
@@ -539,15 +541,23 @@ def estimateAcquisitionTime(detector, scanner=None):
     return et
 
 
-def estimateAutoFocusTime(detector, scanner=None, steps=MAX_STEPS_NUMBER):
+def estimateAutoFocusTime(detector, emt, focus: model.Actuator, dfbkg=None, good_focus=None, rng_focus=None, method=MTD_BINARY) -> float:
     """
-    detector (model.DigitalCamera or model.Detector): Detector on which to
-      improve the focus quality
-    scanner (None or model.Emitter): In case of a SED this is the scanner used
-    Estimates overlay procedure duration
+    Estimates autofocus procedure duration.
+    For the input parameters, see AutoFocus function docstring
+    :return: time in seconds
     """
-    # Add 0.5s per step to account for the focus movement (very roughly approximated)
-    return steps * estimateAcquisitionTime(detector, scanner) + steps * 0.5
+    # adjust to rng_focus if provided
+    rng = focus.axes["z"].range
+    if rng_focus:
+        rng = (max(rng[0], rng_focus[0]), min(rng[1], rng_focus[1]))
+    distance = rng[1] - rng[0]
+    # Optimally, the focus starts from middle to minimum, then maximum. Then it goes back to the middle.
+    # optimistic guess
+    move_time = guessActuatorMoveDuration(focus, "z", distance) + 2 * guessActuatorMoveDuration(focus, "z", distance/2)
+    # pessimistic guess
+    acquisition_time = MAX_STEPS_NUMBER * estimateAcquisitionTime(detector, emt)
```

Comment:
If I understand correctly, you did some tests and the moves where estimated so short that it's almost null? That's why you had to bump up the acquisition time to MAX_STEPS_NUMBER? 
Maybe the moves should be increased? We could consider doing `steps_number` of steps of size `distance/steps_numbers`? Maybe add 10ms per steps, to estimate the overhead? My small experience with the autofocus is that the move of the axis is a large part of the time, so it'd be logical to also estimate it in the same way.